DMTN-148.tex

```diff
@@ -50,9 +50,255 @@
 % ADD CONTENT HERE
 % You can also use the \input command to include several content files.
 
+\section{Intro}
+
+The goal of this tech note is to provide a plan for the handling of
+calibration products in the Gen-3 Butler environment.  There are a
+large number of these products currently in use, but many of them do
+not have a clearly defined process to define, ingest, test, and
+deprecate them.  The lack of documentation on how these processes work
+has led to many implementations of similar paths, further complicating
+the issue.
+
+
+\section{The End State}
+
+The final state that we wish to get to has the following properties:
+
+\begin{itemize}
+\item All calibration products are either lsst.afw.image.Exposure
+  image data or an explicit calibration class in the \verb|ip_isr|
+  package.\footnote{This will require class reorganization to migrate
+    the Defects and Curve calibrations definitions.}
+\item All calibration products should be constructed from a DM Task in
+  the \verb|cp_pipe| package, and have the input data and
+  configuration parameters persisted with the calibration itself.
+  This guarantees that any product can be reconstructed in the future.
+  Exceptions will exist for external or lab created calibrations,
+  although the majority of used calibrations will likely be produced
+  directly.
+\item These provenance, config, and final product files may be saved
+  to an \verb|obs_CAMERA_data| package in git for convenience.  This also allows
+  these files to be migrated to other Butler repositories in a
+  consistent fashion.
+\item All calibration products are accepted for use and have a date
+  range set manually, preventing test runs from being used
+  inadvertently.  Calibrations ingested from git should have sufficient
+  metadata that supply appropriate date ranges.
+\item Calibration information stored in a lsst.afw.cameraGeom object
+  may be used, but is always assumed to be overridden by supplied
+  calibration products.  No new calibration information should be
+  stored in these objects, and currently existing calibration
+  information should be deprecated.
```

Comment:
If this is “the end state”, is there a reason for preserving this deprecated data in perpetuity? Why can't we ultimately get rid of it?

Reply:
Cleaning up camera geom would be great and might be worth an RFC to learn who would object to it.

Reply:
I would like to get rid of it entirely, so you'd only go to cameraGeom to get BBoxes, transforms, and amp->detector->camera->sky relations (the geometry).  I agree that it will need an RFC, but I had planned on waiting until there was a clear alternative location defined (essentially after this calibration rework is stable and on master).